main.tf

```diff
@@ -34,13 +34,27 @@ locals {
   validate_resolver_type_input = (var.resolver_type != null && var.update_delegated_resolver == true) ? tobool("var.resolver_type cannot be set if var.update_delegated_resolver is set to true. Only one type of resolver can be created by VPC.") : true
 }
 
+
+##############################################################################
+# Check if existing vpc id is passed
+##############################################################################
+
+data "ibm_is_vpc" "vpc" {
+  count      = var.create_vpc == false ? 1 : 0
+  identifier = var.existing_vpc_id
+}
+
+locals {
+  vpc_id = var.create_vpc ? ibm_is_vpc.vpc[0].id : var.existing_vpc_id
+}
+
 ##############################################################################
 # Create new VPC
 ##############################################################################
 
 resource "ibm_is_vpc" "vpc" {
   count                       = var.create_vpc == true ? 1 : 0
-  name                        = var.prefix != null ? "${var.prefix}-${var.name}-vpc" : "${var.name}-vpc"
+  name                        = var.name != null ? var.name : "${var.prefix}-vpc"
```

Comment:
This is going to break of name and prefix are both set to null. Perhaps add a check to ensure that one of them is set.